modules/host/negotiaterevisecontract.go

```diff
@@ -288,14 +288,14 @@ func (h *Host) managedRPCReviseContract(conn net.Conn) error {
 
 	// Lock the storage obligation during the revision.
 	h.mu.Lock()
-	err = h.lockStorageObligation(so)
+	err = h.tryLockStorageObligation(so.id())
 	h.mu.Unlock()
 	if err != nil {
 		return err
 	}
 	defer func() {
 		h.mu.Lock()
-		err = h.unlockStorageObligation(so)
+		h.unlockStorageObligation(so.id())
 		h.mu.Unlock()
 		if err != nil {
 			h.log.Critical(err)
```

Comment:
checking the wrong `err` here, since `h.unlockStorageObligation(so.id())` doesn't return an error anymore